litmus-portal/graphql-server/utils/misc.go

```diff
@@ -31,55 +31,53 @@ func RandomString(n int) string {
-	defaultState := false
-	if cluster.AgentNsExists == nil {
-		cluster.AgentNsExists = &defaultState
-	}
-	if cluster.AgentSaExists == nil {
-		cluster.AgentSaExists = &defaultState
-	}
+	var (
+		lines                     []string
+		defaultState              = false
+		AgentNamespace            string
+		ServiceAccountName        string
+		DefaultAgentNamespace     = "litmus"
+		DefaultServiceAccountName = "litmus"
+	)
+
 	file, err := os.Open(template)
 	if err != nil {
 		return []byte{}, err
 	}
 	defer file.Close()
 	scanner := bufio.NewScanner(file)
-	var lines []string
 
-	if !*cluster.AgentNsExists || !*cluster.AgentSaExists {
-		comment := fmt.Sprintf("# This is an auto-generated file. DO NOT EDIT")
-		lines = append(lines, comment)
+	if cluster.AgentNsExists == nil {
+		cluster.AgentNsExists = &defaultState
 	}
-	if !*cluster.AgentNsExists {
-		var nsYaml string
-		if cluster.AgentNamespace != nil {
-			nsYaml = fmt.Sprintf("apiVersion: v1\nkind: Namespace\nmetadata:\n  name: %s\n---", *cluster.AgentNamespace)
-		} else {
-			nsYaml = fmt.Sprintf("apiVersion: v1\nkind: Namespace\nmetadata:\n  name: litmus\n---")
-		}
-		lines = append(lines, nsYaml)
+	if cluster.AgentSaExists == nil {
+		cluster.AgentSaExists = &defaultState
 	}
-	if !*cluster.AgentSaExists {
-		var saYaml string
-		if cluster.Serviceaccount != nil && cluster.AgentNamespace != nil {
-			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: %s\n  namespace: %s\n---", *cluster.Serviceaccount, *cluster.AgentNamespace)
-		} else if cluster.Serviceaccount == nil && cluster.AgentNamespace != nil {
-			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: litmus\n  namespace: %s\n---", *cluster.AgentNamespace)
-		} else if cluster.Serviceaccount != nil && cluster.AgentNamespace == nil {
-			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: %s\n  namespace: litmus\n---", *cluster.Serviceaccount)
-		} else {
-			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: litmus\n  namespace: litmus\n---")
-		}
-		lines = append(lines, saYaml)
+
+	if !*cluster.AgentNsExists && cluster.AgentNamespace != nil && *cluster.AgentNamespace != "" {
+		AgentNamespace = *cluster.AgentNamespace
+	} else {
+		AgentNamespace = DefaultAgentNamespace
 	}
 
+	if !*cluster.AgentSaExists && cluster.Serviceaccount != nil && *cluster.Serviceaccount != "" {
+		ServiceAccountName = *cluster.Serviceaccount
+	} else {
+		ServiceAccountName = DefaultServiceAccountName
+	}
+
+	var (
+		headerStr         = "# This is an auto-generated file. DO NOT EDIT\n"
+		namspaceStr       = "----\napiVersion: v1\nkind: Namespace\nmetadata:\n  name: " + AgentNamespace + "\n"
+		serviceAccountStr = "---\napiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: " + ServiceAccountName + "\n  namespace: " + AgentNamespace + "\n---"
+	)
+
+	saYaml := fmt.Sprintf(headerStr + namspaceStr + serviceAccountStr)
```

Comment:
We need to append the ns and sa yamls only if they don't exist in the user's cluster and not always.

Reply:
Okay, Adding it.